spec/models/piece_spec.rb

```diff
@@ -1,18 +1,232 @@
 require 'rails_helper'
 
 RSpec.describe Piece, type: :model do
-  describe 'piece#valid_move?' do
+
+  describe "piece#move_on_the_board?" do
     before(:each) do
       @game = FactoryGirl.build(:game)
-      @piece = FactoryGirl.create(:piece, file: 4, rank: 4, game: @game)
     end
-    it 'should return true if piece is on the board' do
+
+    it "should return true if piece is on the board" do
       @piece = FactoryGirl.create(:piece, file: 1, rank: 2, game: @game)
-      expect(@piece.valid_move?(1, 3)).to eq(true)
+      expect(@piece.move_on_the_board?(1,3)).to eq(true)
     end
-    it 'should return Not Valid if piece is not on the board' do
+
+    it "should return false if piece is not on the board" do
       @piece = FactoryGirl.create(:piece, file: 1, rank: 2, game: @game)
-      expect(@piece.valid_move?(1, -1)).to eq(false)
+      expect(@piece.move_on_the_board?(1, -1)).to eq(false)
+    end
+  end
+
+  describe "piece#is_capturing?" do
+    before do
+      @game = FactoryGirl.create(:game)
+    end
+
+    it "is capturing" do
+      @piece = FactoryGirl.create(:piece, file: 4, rank: 4, game: @game)
+      @capture_piece = FactoryGirl.create(:piece, file: 5, rank: 5, game: @game)
+
+      result = @piece.is_capturing?(@piece.file, @piece.rank)
```

Comment:
Did you mean `@piece.is_capturing?(@capture_piece.file, @capture_piece.rank)` ?

Reply:
Yes! That is what I meant.  I knew something would escape me eventually. Surprisingly, the test passed.  I'll make this change in a bit.

I follow that format since it's part of arranging the data, performing an action on that data, and making an assertion. (Three A's: arrange, act, assert).

Reply:
You wrote a lot of code for this feature... That's what a pair of fresh eyes is for :)

Interesting I had not yet heard of "arrange, act, assert". Great learning about this :+1: